src/ophyd_async/core/detector.py

```diff
@@ -179,25 +231,95 @@ async def trigger(self) -> None:
             trigger=DetectorTrigger.internal,
         )
         await written_status
-        await self.writer.wait_for_index(
-            indices_written + 1, timeout=self._frame_writing_timeout
-        )
+        end_observation = indices_written + 1
+        await self._observe_writer_indicies(end_observation)
+
+    async def _observe_writer_indicies(self, end_observation: int):
+        async for index in self.writer.observe_indices_written(
+            self._frame_writing_timeout
+        ):
+            for watcher in self._watchers:
+                watcher(..., index, ...)
```

Comment:
I meant to ask you what this actually needs filling out with, since afaik the watchers are generic callables and have no defined signature or structure.

Reply:
I think I found out what was intended by this. I filled it out for kickoff and the fly scanning case and separated off the step scan case for another ticket.